package/generateScss.mjs

```diff
@@ -58,20 +59,89 @@ const generateSCSSModule = async (moduleName, importObj) => {
       }
       generatedScss += `${key}: ${value};\n`;
     });
-    
+  
+  // media.scss
   } else if (moduleName.toLowerCase() === 'media') {
     Object.keys(importObj).forEach((queryName) => {
       const processedQuery = customMediaHelper.process(queryName);
       queryName = queryName.replace('--', '$');
       generatedScss += `${queryName}: '${processedQuery}';\n`;
     });
+  
+  // shadows.scss
+  } else if (moduleName.toLowerCase() === 'shadows') {
+    
+    let mapKeysValues = '';
+    let lightColor = '';
+    let lightStrength = '';
+    let darkColor = '';
+    let darkStrength = '';
+    const entries = Object.entries(importObj);
+    
+    for (let index = 0; index < entries.length; index++) {
+      let [key, value] = entries[index];
+      
+      if (key == '--shadow-color') {
+        lightColor = value;
+      }    
+      if (key == '--shadow-strength') {
+        lightStrength = value;
+      }
+      if (key.includes('color-@media:dark')) {
+        darkColor = value;
+      }    
+      if (key.includes('strength-@media:dark')) {
+        darkStrength = value;
+      }
+      if (key == '--shadow-color' || key == '--shadow-strength' || key.includes('@')) {
+        continue; // skip light and dark for the other loops
+      } 
+
+      key = key.replace('--shadow-', '');
+      if (key.includes('--inner-shadow-')) {
+        key = key.replace('--inner-shadow-', '\'inner-');
+        key = key.replace(/$/, '\'');
+      }
+     
+      value = value.replace(/var\(--(.*?)\)/g, '$$$1');
+      value = value.replace(/hsl/g, 'Hsl')
+      mapKeysValues += `${key}: (${value})`;
+      
+      if (index < entries.length - 1) {
+        mapKeysValues += ',\n '; // Add comma and new line for all entries except the last one
+      }
+    };
     
+    generatedScss += `@use 'sass:map';
+
+$-shadow-color: null;
+$-shadow-strength: null;
+
+@mixin apply-shadows($shadow-color: null, $shadow-strength: null) {
+  $-shadow-color: $shadow-color !global;
+  $-shadow-strength: $shadow-strength !global;
+}
+
+@function shadow($level, $theme: light, $shadow-color: null, $shadow-strength: null) {
+  $shadow-color: $-shadow-color or if($theme == dark, ${darkColor}, ${lightColor});
```

Comment:
i think this local variable will shadow (ha!) the function argument so you'll have to call it something else.

assuming the mixin and global vars get removed, the local variables could be prefixed with `-` (or even `--` because why not)